### PR TITLE
Expose setGlobalThreadCount()

### LINKED ISSUE #25: Support multi-threaded I/O
First of all, thank you for this crate! I've been using it in my project (https://github.com/abusch/rustracer) with great success. 👍 

I'm not super familiar with OpenEXR but I've noticed it supports multi-threaded I/O by using its own worker thread pool. You configure it by calling `setGlobalThreadCount()`. Would it make sense to expose this function? I tried to give it a shot myself to see if it would work, but somehow I can't compile openexr-rs... I'm getting the following:

```
error[E0433]: failed to resolve. Could not find `Build` in `gcc`
  --> openexr-sys/build.rs:79:24
   |
79 |     let mut gcc = gcc::Build::new();
   |                        ^^^^^ Could not find `Build` in `gcc`
error: aborting due to previous error
error: Could not compile `openexr-sys`.
```

### PR DESCRIPTION
**Pull request by norru**
Resurrected https://github.com/cessen/openexr-rs/pull/26 (original work from @abusch)

Fixes https://github.com/cessen/openexr-rs/issues/25

Tested on Ubuntu 18.04

### CODE REVIEW

**Comment by norru** [comment]
https://github.com/cessen/openexr-rs/issues/25#issuecomment-491632614

**Comment by norru** [comment]
@cessen, do you want any changes?

**Comment by norru** [comment]
Hi @cessen @abusch, any feedback on why this hasn't been merged? Is there anything do you want me to do?

File: src/threads.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    let error =
+        unsafe { openexr_sys::CEXR_set_global_thread_count(thread_count as ::std::os::raw::c_int) };
+
+    if error == 0 {
+        Ok(())
+    } else {
+        Err(Error::Generic(String::from(
+            "Unable to set global thread count",
+        )))
```
Comment (Ralith) [inline src/threads.rs]: If this function is fallible (what exceptions does it throw?) the actual error message should be exposed, as in other wrapped functions.

**Review by Ralith — CHANGES_REQUESTED** [review]

File: src/threads.rs (hunk 1 above)
Comment (norru) [reply src/threads.rs]: Makes sense - I'll need to rebase. The exception propagation was broken when I wrote this: https://github.com/cessen/openexr-rs/issues/34.

Comment (Ralith) [reply src/threads.rs]: To be clear, I would like to see confirmation that exceptions might actually be thrown here.

Comment (norru) [reply src/threads.rs]: I've had a quick look but could not see it documented. The declaration is neither `throw()` nor `noexcept` so no guarantee. Will look harder.

Comment (Ralith) [reply src/threads.rs]: That describes most C++ functions that won't ever throw, which is why I'm suspicious. If we want to be paranoid (e.g. bad_alloc is probably theoretically possible) we could abort in the C wrapper on exception and expose an infallible interface.

**Comment by cessen** [comment]
@norru 
Sorry for the radio silence!  My life is a bit hectic at the moment, and I just haven't had the energy to take a look.

@Ralith 
If you have the time/energy to carry through on reviewing this, please feel free to merge it when everything's been addressed.  (I'm pretty sure you have merge rights...?  Let me know if not.)

File: src/threads.rs (hunk 1 above)
Comment (norru) [reply src/threads.rs]: I've rebased and copied the exception pattern used elsewhere. It might look a bit overkill but I am erring on the side of caution.

**Comment by Ralith** [comment]
Remove the version bumps (this is not a breaking change, and we can bump in a separate commit once we have everything we want in) and this can be merged.

**Comment by norru** [comment]
Word.
What version do you want? 0.6.0? 0.6.1?

**Comment by Ralith** [comment]
Just remove those changes entirely and we'll worry about bumping the version when the time comes to release.

**Comment by norru** [comment]
Done

**Pull request merged**